enhancements/installer/openstack-baremetal-compute.md

```diff
@@ -0,0 +1,434 @@
+
+#### Increased performance
+
+User wants to run the Compute Nodes on baremetal machines for increased
+performance. The Control Plane sits on regular Nova instances to leverage the
+flexibility of virtual machines.
+
+In order to distribute workloads between virtual and baremetal compute nodes in
+a mixed environment, OpenShift admins can assign different labels to the
+OpenShift nodes and the end-user will then use them in the selectors for their
+applications.
+
+### Goals
+
+* Enable running OpenShift MachineSets based on Nova flavors matching
+  Ironic-provisioned baremetal Compute Nodes.
+* Provide the same integration with the underlying OpenStack services as in the
+  fully virtualised environment.
+* Baremetal machines can be added and removed from a running OpenShift cluster
+  using the Machine API.
+* Support both `flat` and `neutron` Ironic Network Interfaces
+  - https://docs.openstack.org/ironic/queens/admin/multitenancy.html
+
+### Non-Goals
+
+**Baremetal bootstrap and control plane nodes.**
+
+This proposal focuses on the OpenShift compute nodes only. Deploying or
+migrating the Control plane to BMs is out of scope.
+
+**Baremetal deployments on anything other than an OpenStack tenant**
+
+There are other efforts that provide OpenShift on the bare metal. See
+[Metal³][metal3-io] and the [Baremetal installers](#baremetal-ipi--upi).
+
+These projects allow deploying OpenShift in an environment where all the nodes
+are physical and there is no OpenStack or other cloud layer in-between.
+
+In contrast, this proposal uses the existing OpenStack provider and extends it
+to allow adding baremetal workers if the cloud supports it.
+
+**Attaching floating IPs to the BM workers.**
+
+The OpenStack cluster might make the baremetal nodes accessible to the outside
+world, but exposing them via floating IPs might represent an increase in
+complexity and is currently not an explicit goal.
+
+## Proposal
+
+This enhancement is less about adding a new logical piece, than it is about
+uncovering unknowns in a configuration that might work out-of-the-box in
+selected environments.
+
+Therefore, the implementation consists of:
+* identifying a reference architecture
+* anticipating probable issues
+* testing and fixing errors
+
+### Characteristics of the reference architecture
+
+Prerequisites:
+
+* The OpenStack cloud runs the [Bare Metal (ironic)][openstack-ironic] service
+  to provide baremetal management
+* The OpenStack [Compute (nova)][openstack-nova] and [Networking
+  (neutron)][openstack-neutron] services are configured to serve bare metal
+  nodes using the Nova API
+* All of the additional complexity inherent to baremetal is handled by the IaaS
+  cloud and its administrators:
+    * Enrolling new hardware and discovering its components
+    * Wiping the disks between deployments
+    * Booting the system up from the desired image
+    * Networking
+* The choice between deploying a virtual or baremetal machine is done by
```

Comment:
I wouldn't call this "implementation detail", because it is not part of our implementation. It is rather an assumption we make about the underlying infrastructure.

I am very cautious about making assumptions because it's a fairly new territory for me. Do you think that this line adds confusion?